src/models/projectModel.js

```diff
@@ -0,0 +1,41 @@
+const mongoose = require('mongoose');
+
+const { Schema } = mongoose;
+
+const projectSchema = new Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    description: {
+      type: String,
+      required: true,
+    },
+    users: [
+      {
+        id: {
+          type: Schema.Types.ObjectId,
+          ref: 'User',
+          required: true,
+        },
+        role: {
+          type: String,
+          enum: ['admin', 'user'],
+          default: 'user',
+        },
+      },
+    ],
+    team: {
+      id: {
+        type: Schema.Types.ObjectId,
+        ref: 'Project',
+      },
+    },
```

Comment:
I'm not sure why you create a child object with only id, instead you can do this:
```suggestion
    team: {
      type: Schema.Types.ObjectId,
      ref: 'Project',
    },
```

Reply:
My bad, didn't know that was possible simply like this